app/controllers/v1/users_controller.rb

```diff
@@ -15,12 +15,29 @@ def index
       end
     end
 
+    def enable_totp
+      authenticate_user
+      key = current_user.enable_totp!
+      current_user.save
+      render_success(key: key)
+    end
```

Comment:
So this is good, but I would personally have this just be part of the PATCH endpoint for users where you update your login preference just like you can update any other field - more RESTful that way and works similarly to the rest of the API.